pydatastructs/linear_data_structures/Mutidimensional Array.py

```diff
@@ -0,0 +1,36 @@
+# API to create N dimensional Array
+'''
+Data Members
+1. _ndim - The number of dimensions in the array.
+2. _data - The python list which will store the data in row major order.
+3. _dtype - The data type of the elements.
+
+Methods
+1. __new__(cls, dtype=NoneType, *args, **kwargs)
+2. __getitem__(self, *indices)
+3. __setitem__(self, value, *indices)
+4. fill(self, elem)
+
+'''
+
+class MutiDimensionalArray(object):
```

Comment:
please make sure there is no typo and take care of the failing tests too

Reply:
Its just proposed API, i was asked to propose an API first before the implementation part... Plz review and suggest possible changes in my proposal.
@manaswinidas @czgdp1807

Reply:
I guess APIs are proposed in the issue itself

Reply:
Idk, i dm @czgdp1807 sir,  we decided that i would propose API in comments of a PR but I wasn't able to create a PR just for comments, so I described the working of methods in attached python file. Plz guide me about the workflow, i am quite confused now.

Reply:
The PR is fine, but the code for multidimensional array should be appended to [this file](https://github.com/codezonediitj/pydatastructs/blob/master/pydatastructs/linear_data_structures/arrays.py).

Reply:
`MultiDimensionalArray` should inherit abstract [Array](https://github.com/codezonediitj/pydatastructs/blob/b0c3136dfaa6d8551adfa208bcdc76bb3f95dd4f/pydatastructs/linear_data_structures/arrays.py#L8) class.

Reply:
Okk, thanks. I'll start working on the implementation part

Reply:
> I guess APIs are proposed in the issue itself

Yes, that should have been done but @raksha009 has done that here which is okay. Let's keep this open and push the changes here for the implementation.